utils/kata-manager.sh

```diff
@@ -543,31 +608,72 @@ test_installation()
 
 handle_installation()
 {
-	local kata_version="${1:-}"
-	local containerd_version="${2:-}"
+	local cleanup="${1:-}"
+	[ -z "$cleanup" ] && die "no cleanup value"
 
-	setup
+	local force="${2:-}"
+	[ -z "$force" ] && die "no force value"
+
+	local only_kata="${3:-}"
+	[ -z "$only_kata" ] && die "no only Kata value"
+
+	# These params can be blank
+	local kata_version="${4:-}"
+	local containerd_version="${5:-}"
+
+	setup "$cleanup" "$force"
 
 	handle_kata "$kata_version"
-	handle_containerd "$containerd_version"
+
+	[ "$only_kata" = "false" ] && \
+		handle_containerd \
+		"$containerd_version" \
+		"$force"
 
 	test_installation
 
-	info "$kata_project and $containerd_project are now installed"
+	if [ "$only_kata" = "true" ]
+	then
+		info "$kata_project is now installed"
+	else
+		info "$kata_project and $containerd_project are now installed"
+	fi
 
 	echo -e "\n${warnings}\n"
 }
 
 handle_args()
 {
-	case "${1:-}" in
-		-h|--help|help) usage; exit 0;;
-	esac
+	local cleanup="true"
+	local force="false"
+	local only_kata="false"
+
+	local opt
+
+	local kata_version=""
+	local containerd_version=""
+
+	while getopts "c:fhk:or" opt "$@"
+	do
+		case "$opt" in
+			c) containerd_version="$OPTARG" ;;
+			f) force="true" ;;
+			h) usage; exit 0 ;;
+			k) kata_version="$OPTARG" ;;
+			o) only_kata="true" ;;
+			r) cleanup="false" ;;
+		esac
+	done
+
+	shift $[$OPTIND-1]
 
-	local kata_version="${1:-}"
-	local containerd_version="${2:-}"
+	[ -z "$kata_version" ] && kata_version="${1:-}"
```

Comment:
Q. can this fail with `set -e` ?  for this kind of checks I think the best  option is `-n` and `||` , isn't it? :thinking:

Reply:
I think it's correct, but given the shell's obscure rules I've added `|| true` "just in case" ;)